inet/vibe/inet/url.d

```diff
@@ -419,6 +420,94 @@ struct URL {
 		return cast(NativePath)this.path;
 	}
 
+	/// Decode percent encoded triplets for unreserved or convert to uppercase
+	private string normalize_percent_encoding(const string input)
+	{
+		char[] normalized;
+
+		for (size_t i = 0; i < input.length; i++)
+		{
+			const char c = input[i];
+			if (c == '%')
+			{
+				if (input.length < i + 3)
+					assert(false, "Invalid percent encoding");
+				
+				char conv = cast(char) input[i + 1 .. i + 3].to!ubyte(16);
+				switch (conv)
+				{
+					case 'A': .. case 'Z':
+					case 'a': .. case 'z':
+					case '0': .. case '9':
+					case '-': case '.': case '_': case '~':
+						normalized ~= conv; // Decode unreserved
+						break;
+					default:
+						normalized ~= input[i .. i + 3].toUpper(); // Uppercase HEX
+						break;
+				}
+
+				i += 2;
+			}
+			else
+				normalized ~= c;
+		}
+
+		return normalized.idup;
+	}
+
+	/**
+		Normalizes URL by:
+			- Converting schema and host to lowercase
+			- Removing port if it is the default port for schema
+			- Removing dot segments in path
+			- Converting percent-encoded triplets to uppercase
+			- Adding slash when path is empty
+			- Adding slash to path when path represents a directory
+			- Decoding percent encoded triplets for unreserved characters
+				A-Z a-z 0-9 - . _ ~ 
```

Comment:
This won't look too good in DDOC. Remember that in DDOC, the first paragraph (anything before the first empty line or section definition) is the title. Then the description. Both are optional.

In other words:
```D
/**
  * Title which is a concise description of this function
  *
  * More extended description of what's going on, perhaps corner cases.
  *
  * FooBar:
  *   Will appear under its own section, if "FooBar" means something special for the section.
  *   Note that `Params`, `Returns`, and co are just sections, with a special meaning.
  *
  * If_your_section_name_needs_whitespace:
  *   Then you can just use underscore.
  */
```

So in this example, the documentation will all show up in the title, which is far from concise.
One suggestion would be to use something like:
```D
/**
  * Normalize the content of this `URL` in place
  *
  * Normalization can be used to create a more consistent and human-friendly
  * string representation of the `URL`.
  * The list of transformations applied in the process of normalization is as follows:
  * [...] 
  */
```